### ISSUE TITLE
VoiceOver out of sync with value of Slider

### ISSUE THREAD

**Issue by jamie-unmind** [radix-ui/primitives#2303]
### Current Behavior

When using VoiceOver (MacOS) and keyboard to control Slider UI, the value read out by the screen reader is 1 step out of sync with the current value of the Slider.

https://github.com/radix-ui/primitives/assets/75371176/d9eccad9-8d97-4ab2-ba33-090bfd189846

### Expected behavior

VoiceOver should be in sync and correctly read out the current value selected on the Slider.

### Reproducible example

The behaviour can be observed using the Slider example from radix-ui docs:
https://codesandbox.io/p/sandbox/tw5zv5?file=/App.jsx

### Suggested solution

My suspicion is there is a race condition when using React state to control the value of the slider. Eg: User changes the slider value → causes React to re-render slider with new value → VoiceOver is also triggered by the slider change but uses the old value (before React re-render has completed). However, I have not been able to prove this for certain or create a fix to solve the issue.

### Your environment

| Software         | Name(s) | Version |
| ---------------- | ------- | ------- |
| Radix Package(s) |    @radix-ui/react-slider     |   1.1.0    |
| React            | react    |  18.1.0       |
| Browser          |  Chrome       |    114.0.5735.198     |
| Assistive tech   |     VoiceOver    |    Ventura 13.4     |
| Node             |  node   |   v16.14.2      |
| npm/yarn         |   yarn      |      1.22.4   |
| Operating System |   macOS      |     Ventura 13.4    |
